### DISCUSSION TITLE
Problem and Ways in posting A Jupiter Notebook file to a Quarto Blog.

### DISCUSSION THREAD

**Discussion by Hertzspr** [quarto-dev/quarto-cli#4902] [Q&A] [answered]
I did something in Python in a Jupyter Notebook. 
I converted it to .qmd, and paste it into a post folder in my blog directory. 

The code ran, but when I tried to render it, I was shown this error: 

`[1/2] posts\empty-post\acc_vi_to_q.qmd
ERROR: TypeError: Cannot read properties of null (reading 'bibliography')`

---

Is there an alternative so I can post the jupyter notebook result in my quarto blog? What if I downloaded my jupyter notebook as .html? How can I use the the .html as a blog post? Can the quarto blog post the .html as a post, or do I need to convert .html into .qmd? Is that possible?

Thank you.

**Comment by mcanouil** [comment]
We can't help without the full picture (the file and context).
Could you share a minimal reproducible example or maybe a Git repository?

**Reply by Hertzspr** [reply]
Hi, this is the git repo https://github.com/Hertzspr/husada_plethora_blog. I haven't push the post to git yet.

**Reply by cscheid** [reply]
Thanks for following up, but we need you to push the actual file that causes the error to be available in the git repository.

**Reply by Hertzspr** [reply]
> Thanks for following up, but we need you to push the actual file that causes the error to be available in the git repository.

It's done, in the end i use the actual .ipynb file, and I just added separate metadata file for the yaml. 

The blogpost with jupyter notebook in my quarto blog https://husada-plethora.netlify.app/posts/accenture_virtual_internship/acc_vi_to_q.html?fbclid=IwAR3qpGAaahO8BH9Zg83AP3770n4ux9xfcAQ2huewfQkkOyOtRNGpQsTMFy0

**Reply by cscheid** [reply]
It's great that you solved the problem! The reason I was asking is that this:

```
[1/2] posts\empty-post\acc_vi_to_q.qmd ERROR: TypeError: Cannot read properties of null (reading 'bibliography')
```

is 100% a quarto bug, and we'd love to fix it. So if you're willing to share the version of `acc_vi_to_q.qmd` that caused that crash, we'd be thankful for it!

**Reply by Hertzspr** [reply]
> It's great that you solved the problem! The reason I was asking is that this:
> 
> ```
> [1/2] posts\empty-post\acc_vi_to_q.qmd ERROR: TypeError: Cannot read properties of null (reading 'bibliography')
> ```
> 
> is 100% a quarto bug, and we'd love to fix it. So if you're willing to share the version of `acc_vi_to_q.qmd` that caused that crash, we'd be thankful for it!

Sure! This is the file https://drive.google.com/file/d/1ZPfKESu_GVqIZQSgqFcdnquca1sKU0Nq/view?usp=sharing

**Reply by mcanouil** [reply]
Did you not push the file to your repository?
If so, the commit SHA is enough to go back to the repository state where the bug occurred.

I found the following https://github.com/Hertzspr/husada_plethora_blog/blob/55eabd63f4336c52edd627a597d09a224db113c3/posts/accenture_virtual_internship/acc_vi_to_q.ipynb but no Quarto document in your Git history.

**Reply by Hertzspr** [reply]
Yeah, I didn't push said error in .qmd to github.

**Comment by rgaiacs** [comment, marked as answer]
Hi @Hertzspr. If you want to skill all quarto execution of your Jupyter Notebook, you can export your Jupyter Notebook to plain markdown and Quarto should render it (based on https://quarto.org/docs/websites/#render-targets).

**Reply by mcanouil** [reply]
Here, I believe some pieces are just missing, but we won't know without the repository or a reproducible example.

**Reply by Hertzspr** [reply]
Nice! Thank you! I tried putting in the .md in it works! In the link you mentioned, it said Quarto could render .ipynb directly? At first I thought it didn't work, because I couldn't find the render button in the IDE, but if I open other pages and deployed the blog, it worked, the page from .ipynb was shown nicely.

**Reply by Hertzspr** [reply]
Can I ask a follow up question? How do I add information like date, banner, and tags in the .ipynb so that it shows in the blog? 
![how](https://user-images.githubusercontent.com/95233522/226640889-9f257c5b-1fcd-497f-ae2d-92e5ddbae307.JPG)

the top post is from a rendered .ipynb in my quarto blog.

**Reply by dragonstyle** [reply]
You can add a raw cell as the first cell and place document metadata in that cell, like:

```markdown
---
author: Norah Jones
date: last-modified
categories:
  - blog
---
```

(or if you already have document metadata, that should be used automatically...) HTH!

**Reply by Hertzspr** [reply]
> You can add a raw cell as the first cell and place document metadata in that cell, like:
> 
> ```gfm
> ---
> author: Norah Jones
> date: last-modified
> categories:
>   - blog
> ---
> ```
> 
> (or if you already have document metadata, that should be used automatically...) HTH!

My bad, this is my yaml. 
![yaml](https://user-images.githubusercontent.com/95233522/226647597-f1bfa826-f97f-4495-868f-a8fd740493af.JPG)

What's the reference to create the yaml in .ipynb? In the link bellow, the YAML use " " for the value. 
https://quarto.org/docs/tools/jupyter-lab.html#yaml-front-matter

**Reply by Hertzspr** [reply]
> You can add a raw cell as the first cell and place document metadata in that cell, like:
> 
> ```gfm
> ---
> author: Norah Jones
> date: last-modified
> categories:
>   - blog
> ---
> ```
> 
> (or if you already have document metadata, that should be used automatically...) HTH!

I've tried to follow this exactly but still failing: 
![aaa](https://user-images.githubusercontent.com/95233522/226649815-2bc085c0-f87b-480b-b89c-6b573287480f.JPG)

**Reply by Hertzspr** [reply]
> You can add a raw cell as the first cell and place document metadata in that cell, like:
> 
> ```gfm
> ---
> author: Norah Jones
> date: last-modified
> categories:
>   - blog
> ---
> ```
> 
> (or if you already have document metadata, that should be used automatically...) HTH!

Thanks, I added _metadata.yaml to the post folder now it has things what I want to add

It's done now. https://husada-plethora.netlify.app/posts/accenture_virtual_internship/acc_vi_to_q.html?fbclid=IwAR3qpGAaahO8BH9Zg83AP3770n4ux9xfcAQ2huewfQkkOyOtRNGpQsTMFy0
